cppcache/include/geode/Serializer.hpp

```diff
@@ -215,6 +220,28 @@ inline void writeObject(apache::geode::client::DataOutput& output,
   }
 }
 
+template <typename TObj>
+inline void writeArrayObject(apache::geode::client::DataOutput& output,
+                             const std::vector<TObj>& array) {
+  output.writeArrayLen(array.size());
+  for (auto&& obj : array) {
+    writeObject(output, obj);
+  }
+}
+
+template <typename TObj>
+inline std::vector<TObj> readArrayObject(apache::geode::client::DataInput& input) {
+  std::vector<TObj> array;
+  int len = input.readArrayLen();
+  if (len >= 0) {
+    array.reserve(len);
+    for (auto&& obj : array) {
+      readObject(input, obj);
```

Comment:
Didn't we determine that reserve would not init the vector with n elements so how does this loop work?